internal/parser/reparser.go

```diff
@@ -6,8 +6,10 @@ import (
 )
 
 func (p *Parser) finishReparsedNode(node *ast.Node) {
-	p.currentParent = node
-	node.ForEachChild(p.setParent)
+	node.ForEachChild(func(n *ast.Node) bool {
```

Comment:
ForEachChild is an indirect call, so is not inlinable and escapes its argument; this is why I had to do the goofy parent setting thing in `SetParentsInChildren`. I would suspect that this wouldn't work.